maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * Maintenance script to remove <imap /> tags from articles
+ *
+ * @usage SERVER_ID=203236 php removeImapTags.php --tiles-set-id=1 --verbose --test
+ */
+
+ini_set( "include_path", dirname( __FILE__ )."/../../" );
+
+ini_set('display_errors', 'stderr');
+ini_set('error_reporting', E_ALL);
+
+require_once( dirname( __FILE__ ) . '/../../Maintenance.php' );
+
+class RemoveImapTags extends Maintenance {
+	static protected $verbose = false;
+	static protected $test = false;
+
+	/**
+	 * @var WikiaMaps
+	 */
+	private $maps;
+
+	/**
+	 * @var Array
+	 */
+	private $mapsClientConfig;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Removes from articles <imap /> tags for given tiles' set id.";
+		$this->addOption( 'test', 'Test mode; make no changes', $required = false, $withArg = false, 't' );
+		$this->addOption( 'verbose', 'Show extra debugging output', $required = false, $withArg = false, 'v' );
+		$this->addOption( 'maps-db-host', 'WikiaMaps DB host', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-user', 'WikiaMaps DB user', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-pass', 'WikiaMaps DB pass', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-name', 'WikiaMaps DB name', $required = true, $withArg = true );
+		$this->addOption(
+			'tiles-set-id',
+			'Specify a tiles\' set id which leads to a group of maps which <imap /> tags should get removed',
+			$required = true,
+			$withArg = true,
+			'tsi'
+		);
+	}
+
+	static public function isVerbose() {
+		return self::$verbose;
+	}
+
+	static public function isTest() {
+		return self::$test;
+	}
+
+	/**
+	 * Print the message if verbose is enabled
+	 *
+	 * @param $msg - The message text to echo to STDOUT
+	 */
+	static public function debug( $msg ) {
+		if ( self::isVerbose() ) {
+			echo $msg . PHP_EOL;
+		}
+	}
+
+	static public function isValidInteger( $int ) {
+		return intval( $int ) > 0;
+	}
+
+	static public function isValidCityId( $cityId ) {
+		return self::isValidInteger( $cityId );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)